spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua

```diff
@@ -120,7 +120,10 @@ describe("CP/DP config compat transformations #" .. strategy, function()
         enabled = true,
         config = {
           second = 1,
-          policy = "local",
+          policy = "redis",
```

Comment:
I had to make these changes to make the test pass but I'm a little bit worried right now since we brought up backwards compatibility concerns.

The worry I have is related to the check in line: https://github.com/Kong/kong/blob/51cf38080b3998781b77a3ab55cf327142373d2a/kong/plugins/rate-limiting/clustering/compat/redis_translation.lua#L2

which is used to translate new redis structure into old one. However it is guarded by an if statement that checks whether `policy == 'redis'` and only then does the translation. The idea was to only do the translation if redis is used. However it results with DP receiving structure like this when policy is local.

```
{
  [day] = userdata: NULL
  [fault_tolerant] = true
  [header_name] = userdata: NULL
  [hide_client_headers] = false
  [hour] = userdata: NULL
  [limit_by] = 'consumer'
  [minute] = userdata: NULL
  [month] = userdata: NULL
  [path] = userdata: NULL
  [policy] = 'local'
  [redis] = {
    [database] = 0
    [host] = userdata: NULL
    [password] = userdata: NULL
    [port] = 6379
    [server_name] = userdata: NULL
    [ssl] = false
    [ssl_verify] = false
    [timeout] = 2000
    [username] = userdata: NULL }
  [second] = 1
  [year] = userdata: NULL }
```

As you can see the DP received the new redis structure. It shouldn't use it since it's configured for _"local"_ but I don't know if it won't trip over fields that it did not expect.

There were 2 ways to fix it and given the urgency I decided to go with simpler one but wanted to confirm if that's correct. 

This solution just assumes that DP is able to receive some fields in new schema and it should ignore them.

The other solution is to remove the if guard and always translate new redis structure into the old one even when it's not used - but this way DP won't be surprised with new fields.